gotchas/cypress-testing.md

```diff
@@ -0,0 +1,28 @@
+# Cypress testing gotchas
+
+### Use cy.should() for assertions to ensure cypress retries them
+
+We were encountering random test failures on CI with our Cypress tests. It turns out that Cypress recommends you to wrap your assertions in `.should` instead of just `.then`:
+
+> Note: Prefer .should() with callback over .then() for assertions as they are automatically rerun until no assertions throw within it but be aware of differences.
+> From: https://docs.cypress.io/api/commands/then.html
+
+The differences that you should be aware of are:
+
+> When using a callback function with .should() or .and(), on the other hand, there is special logic to rerun the callback function until no assertions throw within it. You should be careful of side effects in a .should() or .and() callback function that you would not want performed multiple times.
+> From: https://docs.cypress.io/api/commands/should.html#Differences
+
+So make sure that there are no side effects triggered in the callback you pass to `.should`. To summarize:
+
+* If you're asserting something, use `.should` instead of `.then` (make sure there are no side-effects)
+* If you want to work with the subject yielded from the previous command, and you're not asserting anything, use `.then`
+
+### Retry-ability
```

Comment:
The other h3 has an instructive style, I think we should do it here as well, proposal:

> Leverage on cypress' retry ability

Reply:
Sounds good, will change it 👍